### DISCUSSION TITLE
libgit2 does not respect `GIT_CONFIG_NOSYSTEM`, causing test failures

### DISCUSSION THREAD

**Discussion by mlcui-corp** [arxanas/git-branchless#678] [General]
Currently, the `command::test_move::test_move_standalone_create_gc_refs` and `core::test_gc::test_gc_no_init` tests fail on my machine with "Could not find repository main branch" errors.

The root cause is that libgit2, used by `git-branchless`, does not respect the `GIT_CONFIG_NOSYSTEM` used in tests. As a result, `get_main_branch_name` might use the system-wide `init.defaultBranch` config value, which on my machine is `main`, not `master`:

https://github.com/arxanas/git-branchless/blob/2c1a5b910167f47a617bbeecca42cd694dd4968d/git-branchless-lib/src/core/config.rs#L26-L32

There has been some progress to fixing this upstream, see https://github.com/libgit2/libgit2/pull/6399. I don't think this is too actionable by this project - system configurations could have many incompatible configs, and it's infeasible to unset all of them for tests. For example, setting `merge.conflictStyle` to `diff3` causes test failures in the `command::test_navigation::test_navigation_merge` test.

**Comment by arxanas** [comment]
Thanks for investigating!
